### DISCUSSION TITLE
load Original state of object in one storer after another storer already stored a change but did not yet commit

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#675] [Q&A] [answered]
Hi all (and @hg-ms @fh-ms )
I was wondering about the following usecase:
```
create StorageManager (in thread main)
create storer1 (in thread t1)
retrieve, modify and store object X with storer1 (in thread t1)
create storer2 (in thread t2)
retrieve object X (in thread t2)
commit storer1 changes (in thread t1)
...
```
I would like to know if there is a way for storer2 to get the original state of X WITHOUT impacting the object graph.
calling the following in t2:
`storageManager.persistenceManager().createLoader().getObject( ( storageManager.persistenceManager().lookupObjectId(X)));`
will provide the original value but also overrides the object graph with it (similar to the RELOADER class). I am just looking for a way to read the original state without override the data in the object graph...
is there an API for this?
10x

**Comment by hg-ms** [comment, marked as answer]
Hi,
To reload an object without overriding already existing instances is not possible by the current implementation. A custom BinaryLoader implementation might do that, but this is no trivial task.
